units/host_termination_test.go

```diff
@@ -84,8 +91,8 @@ func TestFlaggingDecommissionedHosts(t *testing.T) {
 		Convey("only hosts in the database who are marked decommissioned"+
 			" should be returned", func() {
 
-			// reset the db
 			require.NoError(t, db.ClearCollections(host.Collection), "error clearing hosts collection")
+			setupHostTerminationQueryIndex(t)
```

Comment:
why did you want to have the index in place for these tests?

Reply:
This is required because if you give a hint, the index must exit or the query will fail. We don't have a way of distinguishing a query run in prod from a query run in tests, so there's no way to omit the hint from the query.

Reply:
ah, got it.
Does it work with only the first key ("status" and not "spawn_options.spawned_by_task"), then?

Separately, would an alternative be to only set the hint when `ExecutionEnvironmentType == production`?

Reply:
No, index patterns have to be exact matches.

I would prefer to avoid adding prod-based conditional code execution as much as possible. Switching on prod/non-prod tends to be a hack-ish solution that becomes more kludgy the more you lean on it. If we ever have to introduce more query hints than just this one-off, it might be worth considering doing test setup for indexes somehow.